tests/end_to_end/test_runtimes_e2e.py

```diff
@@ -0,0 +1,31 @@
+from parameterized import parameterized_class
+
+from tests.end_to_end.end_to_end_base import EndToEndBase
+
+
+@parameterized_class(
+    ("runtime", "dependency_manager"),
+    [
+        ("go1.x", "mod"),
+        ("python3.9", "pip"),
+    ],
+)
+class TestHelloWorldDefaultEndToEnd(EndToEndBase):
+    app_template = "hello-world"
+
+    def test_go_hello_world_default_workflow(self):
+        self.default_workflow()
```

Comment:
overall looks good, for purpose this function having an interface since this its something we call within a test, should we pass in a list/dict of assertions?

As-is, all the decisions are handed over to the helper functions, which manages stages, running them and asserting them.

Reply:
Some raw notes:

Maybe we can pass a configuration that is a list of things which knows commands, what kind of assertion to be done, what to assert on and the actual assertion value.

Super raw.

```
Config(
[command="", "assertEquals", "returncode", 0, extraction_function],

extraction function = lambda x: x.process.returncode
```

How to use config:

(getattr(self, attr).getattr(self, attr2), 0)

Discussion points:

* What if we used just one lambda/validation function with custom logic on assertions.